From: Michael Biebl <biebl@debian.org>
Date: Sun, 27 Apr 2014 15:41:26 +0200
Subject: Check for kmod binary

Creating the list of dead device nodes requires kmod. Inside containers
this is not strictly required so we don't want a hard dependency on the
kmod package. If the binary does not exist kmod-static-nodes.service
will fail, so add a condition to check if the binary is available.
---
 units/kmod-static-nodes.service.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/units/kmod-static-nodes.service.in b/units/kmod-static-nodes.service.in
index ff4017b..718d77b 100644
--- a/units/kmod-static-nodes.service.in
+++ b/units/kmod-static-nodes.service.in
@@ -11,6 +11,7 @@ DefaultDependencies=no
 Before=sysinit.target systemd-tmpfiles-setup-dev.service
 ConditionCapability=CAP_MKNOD
 ConditionPathExists=/lib/modules/%v/modules.devname
+ConditionFileIsExecutable=@KMOD@
 
 [Service]
 Type=oneshot
